First draft of new internal cluster with helm charts for timelord and…

oliver-reload opened the pull request:
… harvester.

It's been tested using temp URLS on prod letsencrypt issuer and seems to behaving as expected.

#### What does this PR do?
Adds helm charts for our current internal apps
Adds Terraform to spawn a bare-minimum cluster in GKE (currently on my test-project)
Documentation to come...

#### Where should the reviewer start?
Going through all changes and verifying logic / structure
Optimizations are always welcome

#### How should this be manually tested?
Currently you can visit: https://oliver-the-timelord.reload.dk to test functionality of timelord and that harvester is continuesly updating data.
If you wish to play around with terraform / helm yourself, you can:

1. Authenticate to GKE via an account or service account who is able to perform changes to the project
2. Adjust values in `terraform.tfvars` if you wish to proceed with another project 
3. Intitialise Terraform from the root using: `terraform -chdir=./terraform/internal-utils-alpha init` or cd'ing into the terraform dir first.
4. Configure kubectl: `gcloud container clusters get-credentials $(terraform -chdir=./terraform/internal-utils-alpha output -raw kubernetes_cluster_name) --region $(terraform -chdir=./terraform/internal-utils-alpha output -raw zone)`

#### Any background context you want to provide?
I've used the following parameters with helm to install/upgrade stuff within the cluster:

```
helm upgrade timelord ./helm-charts/timelord \
  --atomic \
  --cleanup-on-fail \
  --install \
  --namespace timelord \
  --create-namespace \
  --set timelord.secret=ProdTimelordSecretHere #GoFetch
```

oliver-reload committed:
First draft of new internal cluster with helm charts for timelord and harvester.
It's been tested using temp URLS on prod letsencrypt issuer and seems to behaving as expected.

achton commented on terraform/internal-utils-alpha/main.tf:
```diff
@@ -0,0 +1,77 @@
+  zone        = var.zone
+}
+
+# Enable GCP APIs
```
This section is commented out. Can be haz removed?

achton commented on helm-charts/harvester/templates/harvester-deployment.yaml:
```diff
@@ -0,0 +1,36 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ .Chart.Name }}-{{ .Release.Name }}-harvester
```
This pattern seems a bit excessive. It will usually translate to `harvester-harvester-harvester`. Maybe just hardcode it or use `.Chart.Name`?

achton commented on helm-charts/harvester/values.yaml:
```diff
@@ -0,0 +1,12 @@
+ingress:
+  host: harvester.reload.dk
+  # enable prod and disable staging to issue REAL certs
+  #issuer: letsencrypt-prod
```
This comment needs a little cleanup.

achton commented on helm-charts/timelord/values.yaml:
```diff
@@ -0,0 +1,8 @@
+# Bump this via helm in a github pipeline instead
+image: ghcr.io/reload/timelord:6643441efe43d3fc0604088af9a51d40077b3045
```
Maybe document how this version bumping will work and link to the repo/pipeline where it happens?

achton commented on terraform/internal-utils-alpha/variables.tf:
```diff
@@ -0,0 +1,35 @@
+  default = "europe-west1-b"
+}
+
+#variable "gcp_service_list" {
```
I guess this is not going to be used?

achton commented on .github/workflows/build-deploy.yml:
```diff
@@ -64,7 +64,7 @@ jobs:
     - name: Deploy helm chart to cluster
       run: |
-        helm upgrade todo-list ./helm-chart/ \
+        helm upgrade todo-list ./helm-charts/todo-list \
```
:+1:

achton reviewed (changes requested):
:clap:

achton commented:
Can you prepare a PR in https://github.com/reload/infrastructure as well?

oliver-reload committed:
Cleaned up comments and adjusted image tags to take input from github workflows
added terraform.tfvars.example to show how to use google credentials with terraform and point to a GCP project

achton reviewed (approved):
*chef's kiss*
